hbase-server/src/main/java/org/apache/hadoop/hbase/util/TableDescriptorChecker.java

```diff
@@ -60,6 +60,17 @@ public final class TableDescriptorChecker {
   private TableDescriptorChecker() {
   }
 
+  private static boolean shouldSanityCheck(final Configuration conf, final TableDescriptor td) {
+    if (conf.getBoolean(TABLE_SANITY_CHECKS, DEFAULT_TABLE_SANITY_CHECKS)) {
+      return true;
+    }
+    String tableVal = td.getValue(TABLE_SANITY_CHECKS);
```

Comment:
should this come before the conf check? Otherwise if you enabled sanity checks globally, then I don't think TableDescriptor could be used to override?

Reply:
I see you sort of pulled this logic from lines 71-79 below, but inverted the conditions. But I think the way you have it here is now more restrictive and doesn't allow the table descriptor to override.